### PR TITLE
Reduce memory usage in conv layer and RoI layer for mobile inference.

### LINKED ISSUE #6657: Reduce memory usage in convolution layer for mobile inference.
Reduce the tempory memory in [ConvOp](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/function/ConvOp.h#L143), we just release this memory in mobile inference.

### LINKED ISSUE #6658: Reduce memory usage in RoI layer for mobile inference.
Do not store the [max index in RoI layer](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/gserver/layers/ROIPoolLayer.h#L45).

### PR DESCRIPTION
**Pull request by qingqing01**
Fix #6658 
Fix #6657

### CODE REVIEW

File: paddle/function/GemmConvOp.cpp [hunk 1]
```diff
@@ -233,6 +233,11 @@ class GemmConvGradInputFunction : public ConvFunctionBase {
       inputGrad += inputChannels * inputHeight * inputWidth;
       outputGrad += outputChannels * outputHeight * outputWidth;
     }
+#ifdef PADDLE_MOBILE_INFERENCE
```
Comment (hedaoyuan) [inline paddle/function/GemmConvOp.cpp]: Should fix `GemmConvFunction` not `GemmConvGradInputFunction`.

Comment (qingqing01) [reply paddle/function/GemmConvOp.cpp]: Oh... I'm sorry. I'll fix it  :)

Comment (qingqing01) [reply paddle/function/GemmConvOp.cpp]: Done.

File: paddle/function/GemmConvOp.cpp [hunk 2]
```diff
@@ -126,6 +126,11 @@ class GemmConvFunction : public ConvFunctionBase {
       inputData += inputChannels * inputHeight * inputWidth;
       outputData += outputChannels * outputHeight * outputWidth;
     }
+#ifdef PADDLE_MOBILE_INFERENCE
+    if (Device == DEVICE_TYPE_CPU) {
+      delete memory_;
```
Comment (hedaoyuan) [inline paddle/function/GemmConvOp.cpp]: Need to use reset to release memory

Comment (qingqing01) [reply paddle/function/GemmConvOp.cpp]: Done in https://github.com/PaddlePaddle/Paddle/pull/6666

**Pull request merged**